### PR TITLE
API headers dependencies removal

### PR DESCRIPTION
**Pull request by est77**
Wrap OIIO and OSL classes used in API headers, replace OSL and OIIO includes in public headers by forward declarations.

With these changes, OIIO and OSL headers are not required anymore to build appleseed.maya.

### CODE REVIEW

File: src/appleseed/renderer/api/rendering.h [hunk 1]
```diff
@@ -45,7 +45,6 @@
 #include "renderer/kernel/rendering/masterrenderer.h"
 #include "renderer/kernel/rendering/nulltilecallback.h"
 #include "renderer/kernel/rendering/progressive/progressiveframerenderer.h"
-#include "renderer/kernel/rendering/scenepicker.h"
```
Comment (dictoon) [inline src/appleseed/renderer/api/rendering.h]: This one should remain I think.

File: src/appleseed/renderer/meta/tests/test_environmentedf.cpp [hunk 2]
```diff
@@ -191,16 +189,17 @@ TEST_SUITE(Renderer_Modeling_EnvironmentEDF)
             TextureStore texture_store(m_scene);
             TextureCache texture_cache(texture_store);
 
-            boost::shared_ptr<OIIO::TextureSystem> texture_system(
-                OIIO::TextureSystem::create(),
-                boost::bind(&OIIO::TextureSystem::destroy, _1));
+            std::shared_ptr<OIIOTextureSystem> texture_system(
+                OIIOTextureSystemFactory::create(),
+                [](OIIOTextureSystem* object) { object->release(); });
```
Comment (dictoon) [inline src/appleseed/renderer/meta/tests/test_environmentedf.cpp]: Lovely!

File: src/appleseed/renderer/api/rendering.h (hunk 1 above)
Comment (est77) [inline src/appleseed/renderer/api/rendering.h]: scenepicker.h brings ShadingPoint and that brings OSL back again.
We cannot remove the OSL includes from shadingpoint. The class holds an instance of OSL::ShaderGlobals.

ScenePicker is not being used in any of our plugins as far as I know. It's only used in studio.

Comment (dictoon) [inline src/appleseed/renderer/api/rendering.h]: The fact that it's used in .studio makes removing this line problematic though. How can it compile?

Comment (est77) [inline src/appleseed/renderer/api/rendering.h]: I added the include in studio where it is used.

**Pull request merged**